src/bgw_policy/interval.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file and its contents are licensed under the Apache License 2.0.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-APACHE for a copy of the license.
+ */
+
+#include <postgres.h>
+#include <funcapi.h>
+#include <utils/typcache.h>
+#include <parser/parse_type.h>
+#include <access/htup_details.h>
+
+#include "bgw_policy/interval.h"
+
+void
+ts_interval_deform_datum(Datum ts_interval_datum, Datum *values, bool *isnull)
```

Comment:
I would think this should return the FormData object instead of filling in the nulls and values. That way the validation logic should be contained in here.

Reply:
I made the interface identical to `heap_deform_tuple` thinking that consistency with the standard postgres type deform API would be more beneficial  but I switched it to what you suggested.